### ISSUE TITLE
max_line_length option doesn't play nice with transforms option

### ISSUE THREAD

**Issue by spudly** [clutchski/coffeelint#449]
If you use a preprocessor to transform your code via coffeelint's "transform" option, you end up with a whole bunch of erroneous warnings/errors about line-length (beause it computes line-length based on transformed code, not original source code).

It sure would be nice if the max_line_length check was performed before the transformation. As it is, this rule is useless when using transforms.

Original Code:
```coffeescript
  statics:
    render: (props, container) ->
      React.render <EnvPropertyMgmtForm {...props} />, container
```

Transformed Code: (using coffee-react-transform)
```coffeescript
  statics:
    render: (props, container) ->
      React.render React.createElement(EnvPropertyMgmtForm, React.__spread({},  props )), container
```

Result
```
Line exceeds maximum allowed length. Length is 99, max is 80. (max_line_length)
```

**Comment by AsaAyers** [comment]
I recommend either disabling the rule completely if you're using transforms or wrapping risky code to disable it.

```
  statics:
    render: (props, container) ->
      # coffeelint: disable=max_line_length
      React.render <EnvPropertyMgmtForm {...props} />, container
      # coffeelint: enable=max_line_length
```

I think it would take some pretty significant changes to make some rules run after the transform and other run after and it has a high risk of producing more inconsistencies.

Ideally someone (else) would figure out a way to have `coffee-react-transform` produce a source map and then have CoffeeLint consume that sourcemap.
